logbucket/logbucket.go

```diff
@@ -142,16 +155,58 @@ func (d *Downloader) downloadObject(obj *s3.Object) error {
 	if err != nil {
 		return fmt.Errorf("Error downloading object file: %s", err)
 	}
-
 	logrus.WithFields(logrus.Fields{
 		"bytes":  nBytes,
 		"file":   f.Name(),
 		"entity": d.String(),
 	}).Info("Successfully downloaded object")
+	
+	// TODO: replace ioutil.ReadFile with a peeker
+	contents, err := ioutil.ReadFile(f.Name())	
+	if contents[0] == 31 && contents[1] == 139 {
```

Comment:
Can you add comments to describe what's happening here? My guess is that it's checking the contents of the file to see if it's zipped in some format. Is there a more idiomatic golang way to do this? If not - adding a comment would be much appreciated.

Reply:
FYI @pravipati @snoozan - If `gunzip`-ing needs to be done with the downloaded objects, it should be done in the `Publisher` for that service, e.g., https://github.com/honeycombio/honeyaws/blob/master/publisher/cloudfront.go#L62-L93, not here. `downloadObjects` simply downloads the object to the local FS and passes the path of the object along the channel. 

By doing the gunzip in `Publisher`, there is no intermediate step to re-write the gunzipped file to disk like there is here (it can be `Scan`-ed as the gzip reader reads), and also the check for gzip magic can probably be avoided altogether since according to [the docs](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html) it's always expected to have a `*.gz` for `elbv2`. I'm not sure why AWS doesn't gzip ELB classic logs. Legacy maybe.